### ISSUE TITLE
undefined reference to `_imp__TF_Version' when compiling in C

### ISSUE THREAD

**Issue by JushBJJ** [tensorflow/tensorflow#34558]
<em>Please make sure that this is a build/installation issue. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:build_template</em>

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Windows 10 x86 Version 10.0.18362.476]
- TensorFlow installed from (source or binary): Binary
- TensorFlow version: Below 2.0
**GCC:**
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=c:/mingw/bin/../libexec/gcc/mingw32/8.2.0/lto-wrapper.exe
Target: mingw32
Configured with: ../src/gcc-8.2.0/configure --build=x86_64-pc-linux-gnu --host=mingw32 --target=mingw32 --prefix=/mingw --disable-win32-registry --with-arch=i586 --with-tune=generic --enable-languages=c,c++,objc,obj-c++,fortran,ada --with-pkgversion='MinGW.org GCC-8.2.0-3' --with-gmp=/mingw --with-mpfr=/mingw --with-mpc=/mingw --enable-static --enable-shared --enable-threads --with-dwarf2 --disable-sjlj-exceptions --enable-version-specific-runtime-libs --with-libiconv-prefix=/mingw --with-libintl-prefix=/mingw --enable-libstdcxx-debug --with-isl=/mingw --enable-libgomp --disable-libvtv --enable-nls --disable-build-format-warnings
Thread model: win32
gcc version 8.2.0 (MinGW.org GCC-8.2.0-3)

**Describe the problem**
Tensorflow functions for C api doesn't work.

**Provide the exact sequence of commands / steps that you executed before running into the problem**

Visited: https://www.tensorflow.org/install/lang_c
Downloaded Binary for Windows (CPU Only): https://storage.googleapis.com/tensorflow/libtensorflow/libtensorflow-cpu-windows-x86_64-1.14.0.zip
Extracted zip file into folder.
Copy and Pasted lib files (including the dll file) into C:\MinGW\lib 
Copy and Pasted include files and folders into C:\MinGW\include

Made new file called "hello_tf.c".
Copy and pasted the example program source code from https://www.tensorflow.org/install/lang_c into the file.

Opened up cmd, went to the file directory and typed:
**gcc hello_tf.c -ltensorflow -o hello_tf.exe && hello_tf.exe**

Output:
![image](https://user-images.githubusercontent.com/36951064/69496153-c6f91380-0f1a-11ea-8be0-80786052dfcb.png)

Basically the same on other functions like TF_DataTypeSize:
![image](https://user-images.githubusercontent.com/36951064/69496162-e8f29600-0f1a-11ea-86a6-a92afa4229a5.png)

Code: 
![image](https://user-images.githubusercontent.com/36951064/69496178-20614280-0f1b-11ea-84cd-ea039b5d1ad1.png)

**Comment by JushBJJ** [comment]
This issue is still ongoing.

**Comment by JushBJJ** [comment]
Update:
This issue is is present in Linux as well, though except it says "invaild instruction", even tested on python. The solution is compiling tensorflow, though I cant be 100% sure because i haven't finished compiling yet, its been 2 days with RAM Usage Issues but dont mind that. If successful with Linux when installed for my CPU I will attempt to try to compile a DLL and lib for Windows and see how it goes.

I have tried v1.15 for C, havent tested python though it is now expected to not work on my CPU unless I compile.

**Comment by gunan** [comment]
What is your CPU make and model?
All tf binaries require CPUs with AVX instruction sets at the minimum.

**Comment by JushBJJ** [comment]
Intel(R) Pentium(R) CPU N4200 @ 1.10GHz, 1101 Mhz, 4 Core(s), 4 Logical Processor(s)

**Comment by gunan** [comment]
Invalid instruction issue is definitely https://github.com/tensorflow/tensorflow/issues/19584
You will have to build libtensorflow from scratch to be able to use it.

@bmzhao Could you try to see if the above example works?

**Comment by JushBJJ** [comment]
Okay.

**Comment by zhilee2019** [comment]
I exactly have the same problem. I also implented the same hello.cpp in Windows with MinGW and came across the same problem. But my CPU is Intel i5-8250u, which has AVX. Can you help me @gunan

**Comment by Reon90** [comment]
Use msvc instead of gcc from mingw. As I can understand mingw doesn't support AVX.
```
cl main.c tensorflow.lib -link -LIBPATH:.
```

**Comment by JushBJJ** [comment]
Will try later.

**Comment by cciprianmihai** [comment]
Hello. Does anybody know how to add `cl` to the path? Because it is not recognized as an internal or external command. Thank you very much!

**Comment by Reon90** [comment]
@cciprianmihai you should open Command Prompt for VS in windows start menu.

**Comment by Saduf2019** [comment]
@JushBJJ 
As tf version below 2.0 is not actively supported, please upgrade to 2.x and let us know if this is still an issue.

**Comment by JushBJJ** [comment]
> @JushBJJ 
> As tf version below 2.0 is not actively supported, please upgrade to 2.x and let us know if this is still an issue.

I've recently changed computers (intel i5 CPU), and Tensorflow works 2.x works for me. This is no longer an issue. Sorry for late response.

**Comment by Saduf2019** [comment]
@JushBJJ
Thank you for your update, glad the issue is resolved for you.

**Comment by YounesZahraoui** [comment]
Does anyone solve this issue ??

**Comment by JushBJJ** [comment]
Wow this was me 5 years ago
